internal/system/food.go

```diff
@@ -0,0 +1,88 @@
+package system
+
+import (
+	"fmt"
+
+	"github.com/arsham/neuragene/internal/asset"
+	"github.com/arsham/neuragene/internal/component"
+	"github.com/arsham/neuragene/internal/entity"
+	"github.com/arsham/neuragene/internal/geom"
+	"github.com/hajimehoshi/ebiten/v2"
+)
+
+// Food system handles the reward that entities get from interaction with food.
+type Food struct {
+	entities   *entity.Manager
+	components *component.Manager
+	assets     *asset.Manager
+	sprite     *ebiten.Image
+}
+
+var _ System = (*Food)(nil)
+
+func (a *Food) String() string { return "Food" }
+
+// setup returns an error if the entity manager or the asset manager is nil.
+func (a *Food) setup(c controller) error {
+	a.entities = c.EntityManager()
+	a.assets = c.AssetManager()
+	a.components = c.ComponentManager()
+	fmt.Print("FRUITY")
+	if a.entities == nil {
+		return fmt.Errorf("%w: entity manager", ErrInvalidArgument)
+	}
+	if a.assets == nil {
+		return fmt.Errorf("%w: asset manager", ErrInvalidArgument)
+	}
+	if a.components == nil {
+		return fmt.Errorf("%w: component manager", ErrInvalidArgument)
+	}
+	a.sprite = a.assets.Sprites()[asset.FruitApple]
+	return nil
+}
+
+// update spawns an ant every 100 frames.
+func (a *Food) update(state component.State) error {
+	if !all(state, component.StateSpawnAnts, component.StateRunning) {
+		return nil
+	}
+	if ebiten.IsMouseButtonPressed(ebiten.MouseButton0) {
+		x, y := ebiten.CursorPosition()
+		err := a.spawnFood(float64(x), float64(y))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+const foodMask = entity.Positioned | entity.BoxBounded | entity.Lifespan
```

Comment:
Will the food going to rot? :laughing: (I like it)

Reply:
I was thinking that. Could be interesting to see if it has any effects on the behaviour of the ants, promoting faster ants maybe?

Reply:
It's a good idea.